test/minikube/minikube_sm_test.go

```diff
@@ -0,0 +1,291 @@
+
+			result, _ := testlib.RunSQL(t, namespaceName, admin0, "demo", "create table user.newtable (id bigint)")
+			t.Log(result)
+		}
+
+		defer func() {
+			status, _ := k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "exec", admin0, "--", "nuocmd", "show", "database", "--db-name", opt.DbName)
+			fmt.Println("database status on EXIT", status)
+		}()
+
+		t.Run("verifyRestartOnLostArchive", func(t *testing.T) {
+			testlib.AwaitDatabaseRestart(t, namespaceName, admin0, "demo", &databaseOptions, restart)
+
+			status, _ := k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "exec", sm0, "--", "ls", "-l", "/var/opt/nuodb/archive/nuodb/demo")
+			t.Log("after archive delete: ", status)
+		})
+	})
+
+	/*
+		t.Run("startDatabaseDaemonSet", func(t *testing.T) {
+			defer testlib.Teardown(testlib.TEARDOWN_DATABASE)
+			databaseOptions := helm.Options{
+				SetValues: map[string]string{
+					"database.sm.resources.requests.cpu":    testlib.MINIMAL_VIABLE_ENGINE_CPU,
+					"database.sm.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+					"database.te.resources.requests.cpu":    testlib.MINIMAL_VIABLE_ENGINE_CPU,
+					"database.te.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+					"backup.persistence.enabled":            "true",
+					"backup.persistence.size":               "1Gi",
+					"database.enableDaemonSet":              "true",
+					// prevent non-backup SM from scheduling
+					"database.sm.nodeSelectorNoHotCopyDS.nonexistantTag": "required",
+					"database.autoRestore.source":                        ":latest",
+				},
+			}
+
+			testlib.StartDatabase(t, namespaceName, admin0, &databaseOptions)
+
+			databaseChartName := testlib.StartDatabase(t, namespaceName, admin0, &databaseOptions)
+
+			opt := testlib.GetExtractedOptions(&databaseOptions)
+			smPodTemplate := fmt.Sprintf("sm-%s-nuodb-%s-%s", databaseChartName, opt.ClusterName, opt.DbName)
+			sm0 := testlib.GetPodName(t, namespaceName, smPodTemplate)
+
+			restart := func() {
+				kubectlOptions := k8s.NewKubectlOptions("", "")
+				kubectlOptions.Namespace = namespaceName
+
+				k8s.RunKubectl(t, kubectlOptions, "exec", sm0, "--", "rm", "-rf", "/var/opt/nuodb/archive/nuodb/demo")
+				testlib.RunSQL(t, namespaceName, admin0, "demo", "select 1 from dual")
+			}
+
+			t.Run("verifyRestartOnLostArchive", func(t *testing.T) {
+				testlib.AwaitDatabaseRestart(t, namespaceName, admin0, "demo", &databaseOptions, restart)
+			})
+		})
+	*/
+
+}
+
+func healDatabase(t *testing.T, namespaceName string, podName string, adminName string, databaseOptions *helm.Options) {
+	opts := testlib.GetExtractedOptions(databaseOptions)
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespaceName)
+
+	restoreArchive(t, namespaceName, podName, "archive", ":latest")
+	err := k8s.RunKubectlE(t, kubectlOptions, "exec", adminName, "--", "nuocmd", "shutdown", "database", "--db-name", opts.DbName)
+	assert.NoError(t, err, "'nuocmd shutdown database' failed")
```

Comment:
you probably want `require.NoError`. assert will not terminate the test here and will go into a 5 min wait on the next line. I am assuming that you want to throw an exception here and terminate the test.

Reply:
Great catch - thank you.

I thought I had found and changed all the assert -> require in the new code...